src/ff1.rs

```diff
@@ -250,18 +250,25 @@ fn generate_s<CIPH: BlockCipher>(ciph: &CIPH, r: &[u8], d: usize) -> Vec<u8> {
     let mut s = Vec::from(r);
     s.reserve(d);
     {
-        let mut j = BigUint::one();
+        let mut j = [0u8; 16];
```

Comment:
I think we could take advantage of u128 here. I tested this locally and it passed the tests. Here is what the function would look like:
```rust
fn generate_s<CIPH: BlockCipher>(ciph: &CIPH, r: &[u8], d: usize) -> Vec<u8> {
    let mut s = Vec::from(r);
    s.reserve(d);
    {
        let mut j = 0u128;
        while s.len() < d {
            j += 1;
            let mut block = j.to_be_bytes();
            for k in 0..16 {
                block[k] ^= r[k];
            }
            ciph.encrypt_block(&mut GenericArray::from_mut_slice(&mut block));
            s.extend_from_slice(&block[..]);
        }
    }
    s.truncate(d);
    s
}
```

Reply:
Heh, nice! I wrote this quite some time ago, and Rust has advanced far enough that this is now feasible with `u128::to_be_bytes()`. I'll push a commit with this change 😄